symbulate/plot.py

```diff
@@ -40,4 +41,117 @@ def plot(*args, **kwargs):
+	return density
+    
+def add_colorbar(fig, type, mappable, label):
+    if 'marginal' not in type:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+    else:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+    cbar = plt.colorbar(mappable=mappable, cax=caxes)
+    caxes.yaxis.set_ticks_position('left')
+    cbar.set_label(label)
+    caxes.yaxis.set_label_position('left')
+    return cbar, caxes
+
+def setup_tile(v, bins, discrete):
+    if not discrete:
+        v_lab = np.linspace(min(v), max(v) + 1, bins)
+        v_pos = range(len(v_lab))
+        v_vect = np.digitize(v, v_lab)
+    else:
+        v_lab = np.unique(v)
+        v_pos = range(len(v_lab))
+        v_map = dict(zip(v_lab, v_pos))
+        v_vect = np.vectorize(v_map.get)(v)
+    return v_vect, v_lab, v_pos
+
+def make_tile(x, y, bins, discrete_x, discrete_y, ax):
+    x_vect, x_lab, x_pos = setup_tile(x, bins, discrete_x)
+    y_vect, y_lab, y_pos = setup_tile(y, bins, discrete_y)
+    nums = len(x_vect)
+    counts = count_var(list(zip(y_vect, x_vect)))
+    intensity = np.zeros(shape=(len(y_lab), len(x_lab)))
         
+    for key, val in counts.items():
+        intensity[key] = val / nums
+    hm = ax.matshow(intensity, cmap='Blues', origin='lower', aspect='auto')
+    ax.xaxis.set_ticks_position('bottom')
+    if not discrete_x: x_lab = np.around(x_lab, decimals=1)
+    if not discrete_y: y_lab = np.around(y_lab, decimals=1)
+    setup_ticks(x_pos, x_lab, ax, 'x')
+    setup_ticks(y_pos, y_lab, ax, 'y')
+    return hm
+
+def setup_ticks(pos, lab, ax, axis):
+    if axis == 'x':
+        ax.set_xticks(pos)
+        ax.set_xticklabels(lab)
+    elif axis == 'y':
+        ax.set_yticks(pos)
+        ax.set_yticklabels(lab)
+
+def make_violin(data, positions, ax, axis, alpha):
+    violins = get_violin(data, positions, ax, axis)
+    for part in violins['bodies']:
+        part.set_edgecolor('black')
+        part.set_alpha(alpha)
+    for part in ('cbars', 'cmins', 'cmaxes', 'cmedians'):
+        vp = violins[part]
+        vp.set_edgecolor('black')
+        vp.set_linewidth(1)
+    return violins
+
+def get_violin(data, positions, ax, axis):
+    values = []
+    if axis == 'x':
+        for i in positions:
+            values.append(data[data[:, 0] == i, 1].tolist())
+    elif axis == 'y':
+        for i in positions:
+            values.append(data[data[:, 1] == i, 0].tolist())
+    violins = ax.violinplot(dataset=values, showmedians=True,
+                            vert=False if axis == 'y' else True)
+    setup_ticks(np.array(positions) + 1, positions, ax, axis)
+    return violins
+
+
+def marginal_impulse(count, height, color, ax_marg, alpha, axis):
+    key = list(count.keys())
+    val = list(height)
+    tot = sum(val)
+    val = [i / tot for i in val]
+    if axis == 'x':
+        ax_marg.vlines(key, 0, val, color=color, alpha=alpha)
+    elif axis == 'y':
+        ax_marg.hlines(key, 0, val, color=color, alpha=alpha)
+
+def density2D(x, y, ax):
```

Comment:
Should this be called `make_density2D` to be consistent with the other function names?